### ISSUE TITLE
worksheet links can become stale or unavailable and should be manually fixed

### ISSUE THREAD

**Issue by NyakudyaA** [kartoza/projecta#1326]
# Problem

All lessons have further reading sections where we put links. When the links become unavailable the reading section becomes stale.

# Solution

I suggest a little tool or script that basically generates all invalid links in a PDF/text file like

```
The following links are unavailable
https://changelog.qgis.org/en/qgis/lesson/symbology-22/detail/81/?q=3.1 has invalid links or unavailable links:
* http://foo.bar.txt
```
Or just saying
```
The following sections have invalid links
https://changelog.qgis.org/en/qgis/lesson/symbology-22/detail/81/?q=3.1
https://changelog.qgis.org/en/qgis/lesson/symbology-22/detail/167/?q=3.3
```

This tool can be run per need and should ideally be visible to the lessons manager.

Use case:
Before each training session, we run the tool and manually fix the links in the respective sections

**Comment by NyakudyaA** [comment]
The tool doesn't seem to be working on the deployed instance @sumandari 
![invalid-link](https://user-images.githubusercontent.com/2510900/116858246-5f122680-abfe-11eb-9a70-357ef157762d.png)

**Comment by sumandari** [comment]
Hi @NyakudyaA, 
When I run it on my local machine, the function will take about 10minutes to return the value.
I have discussed it with @lucernae, it seems the time for waiting the return is not feasible to make it as an html endpoint.
We might create a job to call this function.
cc @dimasciput

**Comment by sumandari** [comment]
the fix has been deployed in version 2.1.6

**Comment by sumandari** [comment]
The fixed has been deployed, I will close this ticket now
